shared/tests/utility/motion/TestSplines.cpp

```diff
@@ -198,12 +199,12 @@ TEST_CASE("Test Polynom", "[utility][motion][splines][Polynom]") {
     for (size_t i = 0; i < P.size(); i++) {
         // Get the coefficients for the test polynomial and the true differential values for the given x
         std::vector<long int> coefs = {P[i][5], P[i][4], P[i][3], P[i][2], P[i][1], P[i][0]};
-        Eigen::Vector4i result_true(P[i][7], P[i][8], P[i][9], P[i][10]);
-        int x = P[i][6];
+        Eigen::Matrix<long int, 4, 1> result_true(P[i][7], P[i][8], P[i][9], P[i][10]);
+        long int x = P[i][6];
 
         // Create the polynomial from test coefficients and get the computed differential values for the given x
         utility::motion::splines::Polynom poly(coefs);
-        Eigen::Matrix<int long, 4, 1> result_calc(poly.pos(x), poly.vel(x), poly.acc(x), poly.jerk(x));
+        Eigen::Matrix<long int, 4, 1> result_calc(poly.pos(x), poly.vel(x), poly.acc(x), poly.jerk(x));
```

Comment:
Should these not be `size_t`?

Reply:
I don't think we want to be doing unsigned arithmetic here. These aren't indices

Reply:
You really don't want to be doing unsigned arithmetic anywhere ever indices or not, it's so bug-prone and gives you literally nothing except a higher max value in exchange. And if you need to deal with numbers bigger than an int64... that sounds insane